### PR TITLE
[bug fix]: Fixes bug where charms with multiple units couldn't request certificates

### PR DESCRIPTION
**Pull request by gruyaume**
# Description

- Fixes bug where charms with multiple units couldn't request certificates
- Improves code structure
- Reverts fix for ca chains being lists. This will be re-added later.
- Removes the broken "certificate revoked event". Those won't be able to exist since from the tls certificates requirer point of view, it is impossible to differentiate whether a certificate was revoked or wasn't yet created.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/charms/tls_certificates_interface/v1/tls_certificates.py
Comment (sanchezfdezjavier) [inline lib/charms/tls_certificates_interface/v1/tls_certificates.py:821]: Is there any reason why we don't merge both conditions?
```suggestion
           if certificate and certificate_dict["certificate"] == certificate:
```

File: lib/charms/tls_certificates_interface/v1/tls_certificates.py
Comment (sanchezfdezjavier) [inline lib/charms/tls_certificates_interface/v1/tls_certificates.py:824]: Same as above, why don't we merge both conditions?

File: lib/charms/tls_certificates_interface/v1/tls_certificates.py
Comment (sanchezfdezjavier) [inline lib/charms/tls_certificates_interface/v1/tls_certificates.py:952]: A bit more pythonic?
```suggestion
            list_of_csrs.extend(csr["certificate_signing_request"] for csr in self._requirer_csrs(unit))
```

File: lib/charms/tls_certificates_interface/v1/tls_certificates.py
Comment (sanchezfdezjavier) [inline lib/charms/tls_certificates_interface/v1/tls_certificates.py:1192]: Should we consider using `datetime.now(timezone.utc)` instead of `datetime.utcnow()`? A [warning in the docs](https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow) explains why.
> Warning Because naive datetime objects are treated by many datetime methods as local times, it is preferred to use aware datetimes to represent times in UTC. As such, the recommended way to create an object representing the current time in UTC is by calling datetime.now(timezone.utc).

**Review by sanchezfdezjavier — CHANGES_REQUESTED** [review]

File: lib/charms/tls_certificates_interface/v1/tls_certificates.py
Comment (gruyaume) [reply lib/charms/tls_certificates_interface/v1/tls_certificates.py:1192]: This doesn't seem to work. The datetime taken from the `load_pem_x509_certificate` is not timezone aware while the one using `datetime.now(timezone.utc)` is and those 2 quantities can't be subsctracted.

File: lib/charms/tls_certificates_interface/v1/tls_certificates.py
Comment (gruyaume) [reply lib/charms/tls_certificates_interface/v1/tls_certificates.py]: Done

File: lib/charms/tls_certificates_interface/v1/tls_certificates.py
Comment (gruyaume) [reply lib/charms/tls_certificates_interface/v1/tls_certificates.py]: Done

File: lib/charms/tls_certificates_interface/v1/tls_certificates.py
Comment (gruyaume) [reply lib/charms/tls_certificates_interface/v1/tls_certificates.py]: Done

File: lib/charms/tls_certificates_interface/v1/tls_certificates.py
Comment (ghislainbourgeois) [reply lib/charms/tls_certificates_interface/v1/tls_certificates.py:1194]: In this particular case, this will not be an issue, but I think it is still best to follow the recommendation, at least so that we get used to this construct.

**Review by ghislainbourgeois — APPROVED** [review]
LGTM once the comments already here are resolved.

File: lib/charms/tls_certificates_interface/v1/tls_certificates.py
Comment (gruyaume) [reply lib/charms/tls_certificates_interface/v1/tls_certificates.py:1194]: @ghislainbourgeois Like I mentioned above, I don't see how that's doable in our case.

**Review by sanchezfdezjavier — APPROVED** [review]
LGTM

**Comment by sed-i** [comment]
Is this something that could be captured in a unit test?

**Pull request merged**
